assets/router/deployment.yaml

```diff
@@ -15,11 +15,22 @@ spec:
     metadata:
       labels:
         app: router
+        ingress.openshift.io/component: ingress-controller
```

Comment:
I realized as part of this work that our `app: router` label is extremely vague. I didn't want to update all the other references as part of this PR, so I did an additive change, and we could do a followup to convert other usages.